pkg/replication/replication_test.go

```diff
@@ -1 +1,117 @@
 package replication
+
+import (
+	"sync"
+	"testing"
+
+	"time"
+
+	"github.com/square/p2/pkg/logging"
+
+	"github.com/square/p2/pkg/health"
+	"github.com/square/p2/pkg/health/checker/test"
+	"github.com/square/p2/pkg/kp"
+	"github.com/square/p2/pkg/kp/kptest"
+	"github.com/square/p2/pkg/manifest"
+	"github.com/square/p2/pkg/types"
+)
+
+// TODO mark as slow test so these don't flake travis
+func TestEnactHappyPath(t *testing.T) {
+	errCh := make(chan error)
+	defer close(errCh)
+
+	r := newTestReplication(errCh)
+	r.Enact()
+
+	intent, _, err := r.store.AllPods(kp.INTENT_TREE)
+	if err != nil {
+		t.Fatalf("Encountered error while fetching intent: %v\n", err)
+	}
+
+	if len(intent) != 2 {
+		t.Errorf("Expected to have 2 intent records scheduled but got %d.\n%v", len(intent), intent)
+	}
+	reality, _, err := r.store.AllPods(kp.REALITY_TREE)
+	if err != nil {
+		t.Fatalf("Encountered error while fetching reality: %v\n", err)
+	}
+	if len(reality) != 2 {
+		t.Errorf("Expected to have 2 reality records scheduled but got %d.\n%v", len(reality), reality)
+
+	}
+}
+
+func TestEnactCancellation(t *testing.T) {
+	errCh := make(chan error)
+	defer close(errCh)
+	r := newTestReplication(errCh)
+	r.active = 1
+	r.rateLimiter = time.NewTicker(2 * time.Second) // we need time to cancel
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		r.Enact()
+		wg.Done()
+	}()
+	r.Cancel()
+	wg.Wait()
```

Comment:
if this test was failing this would probably stall forever right? maybe instead of using a waitgroup do a send on a channel inside a `select` with a `time.After()`?